app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java

```diff
@@ -80,4 +82,10 @@ public void login()
         assertNull(database.login("email46@email.com", "abc123"));
     }
 
+    @Test
+    public void insertKnownPost(){
```

Comment:
You need way more tests then this, test all kinds of input and make sure it is what you expect it to be.